RecoTracker/TrackProducer/plugins/TrackRefitter.cc

```diff
@@ -136,6 +136,7 @@ void TrackRefitter::produce(edm::Event& theEvent, const edm::EventSetup& setup)
       theAlgo.runWithVertex(theG.product(), theMF.product(), *theTCollectionWithConstraint, 
 			    theFitter.product(), thePropagator.product(), theBuilder.product(), bs, algoResults);      
       }catch (cms::Exception &e){ edm::LogError("TrackProducer") << "cms::Exception caught during theAlgo.runWithVertex." << "\n" << e << "\n"; throw; }
+      break;
```

Comment:
This looks like as a bug fix: which effect do you expect from it? Why there are no differences in the jenkins outputs? This plugin is apparently rather widely used in cmssw: http://cmslxr.fnal.gov/search?_filestring=&_string=TrackRefitter&_casesensitive=1

While the fix itself looks rather reasonable, do you have an example of the issue you want to fix with it?

Reply:
case Vertex has never been used in CMSSW because there has been no functional VertexConstraintProducer in CMSSW.
The fix here is needed so that upon entering the switch statement, we don't accidentally go into trackParameters case either. Otherwise, it tries to go into theAlgo.runWithTrackParameters, and since constaints are of different type, runs into a segmentation fault.